### DISCUSSION TITLE
Compose Multiplatform support

### DISCUSSION THREAD

**Discussion by mxrsoon** [sceneview/sceneview#559] [Ideas]
Thanks for the library as it is!
Are there any plans to add support for Compose Multiplatform in the future?

**Comment by ThomasGorisse** [comment]
SceneView uses **native renderers per platform** rather than Compose Multiplatform for the rendering layer:

- **Android** — Jetpack Compose + Filament (`io.github.sceneview:sceneview:4.0.0-rc.1`)
- **iOS/macOS/visionOS** — SwiftUI + RealityKit (`SceneViewSwift` via SPM)
- **Web** — Kotlin/JS + Filament.js (`sceneview-web` on npm)
- **Flutter** — PlatformView bridge to native SceneView
- **React Native** — Fabric bridge to native SceneView

Shared logic (math, collision, geometry, animations) lives in `sceneview-core`, a Kotlin Multiplatform module. Rendering stays native because RealityKit is required for visionOS spatial computing, and each platform benefits from native tooling/debugging.

**Comment by rkreienbuehl** [comment]
Would love to see this
